Added retries for downloading the Gradle distribution (#36556)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

ADD RELEASE FEATURES BELOW
vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -->

## New features and usability improvements

> ### Wrapper task retries downloading distribution

Reviewer: ```suggestion
### Automatic retries for wrapper downloads
```

The Gradle Wrapper now automatically retries downloading the distribution if the initial attempt fails.
The improves the resilience of the wrapper against transient network issues. The number of retries and

> the timeout between attempts can be configured via new properties in the gradle-wrapper.properties file.

Reviewer: ```suggestion
By default, the Wrapper will attempt up to 3 retries with a 1-second delay between each.
```

> More details can be found in the dedicated section of the Gradle Wrapper user manual chapter.

Reviewer: ````suggestion
Users can fine-tune the retry logic by adding the following to their Gradle properties file:

```properties
networkRetries=5
networkRetryDelayMs=2000
```
````

ADD RELEASE FEATURES BELOW
vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -->

## New features and usability improvements

### Automatic retries for wrapper downloads

The Gradle Wrapper now supports automatic retries for distribution downloads.

> This reduces build failures caused by unstable network connections or server-side timeouts.

Reviewer: ```suggestion
This helps reduce build failures caused by unstable network connections or temporary server issues.
```
